activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -67,7 +67,7 @@ def sql_color(sql)
       case sql
         when /\A\s*rollback/mi
           RED
-        when /\s*.*?select .*for update/mi, /\A\s*lock/mi
+        when /select .*for update/mi, /\A\s*lock/mi
```

Comment:
I certainly don't object to removing the entire special case.

However, if you decide to keep it, since the portion of the Regexp that can take a huge amount of time to match can also match zero characters, eliminating just that portion&mdash;again, unless I'm missing something&mdash;shouldn't affect whether there's a match or not for any given input.

Either way, I hope something is done so that we can turn our logging level back to debug for tests (and probably production).